src/realm/sync/changeset_encoder.cpp

```diff
@@ -229,7 +229,9 @@ InternString ChangesetEncoder::intern_string(StringData str)
         size_t index = m_intern_strings_rev.size();
         // FIXME: Assert might be able to be removed after refactoring of changeset_parser types?
         REALM_ASSERT_RELEASE_EX(index <= std::numeric_limits<uint32_t>::max(), index);
-        it = m_intern_strings_rev.insert({std::string{str}, uint32_t(index)}).first;
+        bool inserted;
+        std::tie(it, inserted) = m_intern_strings_rev.insert({str, uint32_t(index)});
```

Comment:
I think this is dangerous. When using std::string, we take a copy of the string as we cannot be sure about the lifetime of the string `str` points to.

Reply:
@jbreams flagged the same to me. I will revert the change and use a std::string_view above in m_intern_strings_rev.find()